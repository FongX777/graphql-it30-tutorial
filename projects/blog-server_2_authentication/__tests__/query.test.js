const { gql } = require('apollo-server');
const { createTestClient } = require('apollo-server-testing');
const { constructTestServer } = require('./__utils');

const GET_ME = gql`
  query me {
    me {
      id
      name
      email
      posts {
        id
        title
        body
      }
    }
  }
`;

const GET_USERS = gql`
  query userList {
    users {
      id
      name
      email
      posts {
        id
        title
        body
        author {
          id
          name
        }
        likeGivers {
          id
          name
        }
      }
    }
  }
`;

const GET_USER = gql`
  query user {
    user(id: 1) {
      id
      name
      email
      posts {
        id
        title
        body
      }
    }
  }
`;

const GET_POSTS = gql`
  query postList {
    posts {
      id
      title
      body
      author {
        id
        name
      }
      likeGivers {
        id
        name
      }
    }
  }
`;

const GET_POST = gql`
  query post {
    post(id: 1) {
      id
      title
      body
      author {
        id
        name
      }
      likeGivers {
        id
        name
      }
    }
  }
`;

describe('Queries', () => {
  let query;
  let server;
  beforeEach(() => {
    server = constructTestServer().server;
    query = createTestClient(server).query;
  });

  it('fetches the current user', async () => {
    const res = await query({ query: GET_ME });
    expect(res).toMatchSnapshot();
  });

  it('fetches all users', async () => {
    const res = await query({ query: GET_USERS });
    expect(res).toMatchSnapshot();
  });

  it('fetches single user', async () => {
    const res = await query({ query: GET_USER });
    expect(res).toMatchSnapshot();
  });

  it('fetches all posts', async () => {
    const res = await query({ query: GET_POSTS });
    expect(res).toMatchSnapshot();
  });

  it('fetches single post', async () => {
    const res = await query({ query: GET_POST });
    expect(res).toMatchSnapshot();
  });
});
