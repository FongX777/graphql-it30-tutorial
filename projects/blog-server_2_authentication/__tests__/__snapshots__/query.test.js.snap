// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches all posts 1`] = `
Object {
  "data": Object {
    "posts": Array [
      Object {
        "author": Object {
          "id": "1",
          "name": "Fong",
        },
        "body": "This is my first post",
        "id": "1",
        "likeGivers": Array [
          Object {
            "id": "2",
            "name": "Kevin",
          },
        ],
        "title": "Hello World",
      },
      Object {
        "author": Object {
          "id": "2",
          "name": "Kevin",
        },
        "body": "Hello My Friend!",
        "id": "2",
        "likeGivers": Array [],
        "title": "Nice Day",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches all users 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "email": "fong@test.com",
        "id": "1",
        "name": "Fong",
        "posts": Array [
          Object {
            "author": Object {
              "id": "1",
              "name": "Fong",
            },
            "body": "This is my first post",
            "id": "1",
            "likeGivers": Array [
              Object {
                "id": "2",
                "name": "Kevin",
              },
            ],
            "title": "Hello World",
          },
        ],
      },
      Object {
        "email": "kevin@test.com",
        "id": "2",
        "name": "Kevin",
        "posts": Array [
          Object {
            "author": Object {
              "id": "2",
              "name": "Kevin",
            },
            "body": "Hello My Friend!",
            "id": "2",
            "likeGivers": Array [],
            "title": "Nice Day",
          },
        ],
      },
      Object {
        "email": "mary@test.com",
        "id": "3",
        "name": "Mary",
        "posts": Array [],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches single post 1`] = `
Object {
  "data": Object {
    "post": Object {
      "author": Object {
        "id": "1",
        "name": "Fong",
      },
      "body": "This is my first post",
      "id": "1",
      "likeGivers": Array [
        Object {
          "id": "2",
          "name": "Kevin",
        },
      ],
      "title": "Hello World",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches single user 1`] = `
Object {
  "data": Object {
    "user": Object {
      "email": "fong@test.com",
      "id": "1",
      "name": "Fong",
      "posts": Array [
        Object {
          "body": "This is my first post",
          "id": "1",
          "title": "Hello World",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches the current user 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "fong@test.com",
      "id": "1",
      "name": "Fong",
      "posts": Array [
        Object {
          "body": "This is my first post",
          "id": "1",
          "title": "Hello World",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
